cmd/segment-reaper/observer.go

```diff
@@ -143,11 +145,88 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
+				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
+					// TODO should we also check if its valid sequence
+					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+					brokenObject = true
+				}
+
+				if brokenObject {
+					err := obsvr.printSegment(ctx, "l", bucket, path, obsvr.writer)
+					if err != nil {
+						return err
+					}
+				}
+			} else {
+				brokenObject = true
+			}
+
+			if brokenObject {
+				for index := 0; index < maxNumOfSegments; index++ {
+					has, err := object.segments.Has(index)
+					if err != nil {
+						return nil
```

Comment:
We can call panic for this case, it should only happen if we have a bug.

Reply:
panic call added